vshard/storage/init.lua

```diff
@@ -1085,6 +1085,22 @@ local function rebalancing_is_in_progress()
     return f ~= nil and f:status() ~= 'dead'
 end
 
+local function recovery_bucket_stat(bid)
```

Comment:
Check the commit message, pls:
```
    The new function is used by bucket recovery instead of
    bucket_stat(). The problem of the latter is that it allows to be
    invoked on a non-master instance which is not ok for recovery - it
    needs to get the most up to date state of the bucket.
    
    On the other hand it wasn't possible to just ban bucket_stat() on
    non-master instances, because it is a public function and it was
    never documented to work exclusively on master instances.
```

Reply:
I meant the following:

```diff
diff --git a/vshard/storage/init.lua b/vshard/storage/init.lua
index d9a8557..805bb0a 100644
--- a/vshard/storage/init.lua
+++ b/vshard/storage/init.lua
@@ -1125,15 +1125,7 @@ local function recovery_bucket_stat(bid)
     if not ok then
         return nil, err
     end
-    local b = box.space._bucket:get{bid}
-    if b == nil then
-        return
-    end
-    return {
-        id = bid,
-        status = b.status,
-        is_transfering = M.rebalancer_transfering_buckets[bid],
-    }
+    return bucket_stat(bid)
 end
 
 --
```

Reply:
Ah, I see. The reason is that I don't like `bucket_stat()` for internal purposes. It is too verbose and is doing too much.